pandas/io/parquet.py

```diff
@@ -180,8 +180,12 @@ def write(
             mode="wb",
             is_dir=partition_cols is not None,
         )
-        if isinstance(path_or_handle, io.BufferedWriter):
-            path_or_handle = path_or_handle.raw.name
+        if (
+            isinstance(path_or_handle, io.BufferedWriter)
+            and isinstance(path_or_handle.raw, io.FileIO)
```

Comment:
Why does `.raw` need to be an instance of `io.FileIO`, your current code doesn't use `.raw`? Wouldn't `isinstance(path_or_handle, io.FileIO) and isinstance(path_or_handle.name, (str, bytes))` be sufficient?

Reply:
`io.BufferedWriter` wraps `io.FileIO` using the base class `io.BufferedIOBase`. We are able to access the io.FileIO class using `.raw` attribute . I have now added a condition to check if the wrapped class is `io.FileIO` and also check if the `.name` attribute is of type `str` or `bytes`.

When I directly check for `isinstance(path_or_handle, io.FileIO)` instead of `isinstance(path_or_handle.raw, io.FileIO)`, the condition returns `false` when `file_or_path` is a `io.BufferedWriter` object

Reply:
Thanks, I assumed that most file handles would inherit from `io.FileIO` and not `io.BufferedWriter`.

Since you directly access `path_or_handle.name` (and not `.raw.name`), I would suggest:
```py
 if isinstance(path_or_handle, io.BufferedWriter) and hasattr(path_or_handle, "name") and isinstance(path_or_handle.name, (str, bytes)):
    ...
```
or
```py
 if isinstance(path_or_handle, io.BufferedWriter) and isinstance(getattr(path_or_handle, "name", None), (str, bytes)):
    ...
```

Reply:
Thanks, I have now updated the condition to `hasattr(path_or_handle, "name")` from `isinstance(path_or_handle.raw, io.FileIO)`